packages/editor-ui/src/middleware/enterprise.ts

```diff
@@ -0,0 +1,30 @@
+import type { RouterMiddleware } from '@/types/router';
+import { useSettingsStore } from '@/stores';
+import type { EnterpriseEditionFeature } from '@/constants';
+import { VIEWS } from '@/constants';
+
+export type EnterpriseMiddlewareOptions = {
+	features: EnterpriseEditionFeature[];
+	mode?: 'oneOf' | 'allOf';
+};
+
+export const enterprise: RouterMiddleware<EnterpriseMiddlewareOptions> = async (
+	to,
+	from,
+	next,
+	options,
+) => {
+	const settingsStore = useSettingsStore();
+	const mode = options.mode ?? 'allOf';
+
+	let valid: boolean;
+	if (mode === 'allOf') {
+		valid = options.features.every((feature) => settingsStore.isEnterpriseFeatureEnabled(feature));
```

Comment:
just a nitpick (coming from my functional programming self :))
it could be pointfree

```suggestion
		valid = options.features.every(settingsStore.isEnterpriseFeatureEnabled);
```

Reply:
Done!